x-pack/plugins/observability/public/context/has_data_context.tsx

```diff
@@ -147,7 +147,9 @@ export function HasDataContextProvider({ children }: { children: React.ReactNode
   });
 
   const hasAnyData = (Object.keys(hasDataMap) as ObservabilityFetchDataPlugins[]).some(
-    (app) => hasDataMap[app]?.hasData === true
+    (app) =>
+      hasDataMap[app]?.hasData === true ||
+      (Array.isArray(hasDataMap[app]?.hasData) && (hasDataMap[app]?.hasData as Alert[])?.length > 0)
```

Comment:
Nit: Maybe let's assign `hasDataMap[app]?.hasData` to a `const` so we don't have to repeat it?

```suggestion
    (app) => {
      const appHasData = hasDataMap[app]?.hasData;
      return appHasData === true ||
       (Array.isArray(appHasData) && (appHasData as Alert[])?.length > 0)
     }
```

Reply:
@claudiopro / @Zacqary do you think we could do this logic here https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/context/has_data_context.tsx#L126?

Reply:
Although if we change it where I pointed, we'd have to change the way we get the alerts here too, https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/overview/index.tsx#L75:L76

Reply:
Sounds reasonable! Done